src/eventstream.cr

```diff
@@ -0,0 +1,56 @@
+require "base64"
+
+module EventStream
+    # Each message has this structure:
+    #
+    #   total_length   : Int32  (4 bytes)
+    #   headers_length : Int32  (4 bytes)
+    #   prelude_crc    : Int32  (4 bytes)
+    #   headers        : Bytes  (headers_length bytes)
+    #   payload        : Bytes  (total_length - headers_length - 16 bytes)
+    #   message_crc    : Int32  (4 bytes)
+    #
+    # Note: We read the raw headers block here but do not parse individual header fields
+    # or verify CRC.
+
+    struct EventMessage
+        getter total_length     : Int32
+        getter headers_length   : Int32
+        getter headers          : String
+        getter payload          : String
+        getter message_crc      : Int32
+
+        def initialize(@total_length : Int32, @headers_length : Int32, @headers : String, @payload : String, @message_crc : Int32)
+            @total_length = total_length
+            @headers_length = headers_length
+            @headers = headers
+            @payload = payload
+            @message_crc = message_crc
```

Comment:
These are handled by the arguments, so this method can be empty. One of my favorite things about Crystal constructors! 🎉

```suggestion
```